design-documents/module-split.md

```diff
@@ -0,0 +1,83 @@
+# Module split
+
+## About
+
+This document clarifies the following two questions of the original [service isolation proposal](service-isolation.md)
+1. What new modules should be called
+2. What to do with module namespaces, keep old or change so they match new module name
+
+## Design
+
+1. Module naming
+
+    The following is proposed naming for modules:
+    API (service contracts)
+    * CatalogApi _recommended_
+    
+    The following is proposed naming for modules:
+    Admin Ui and admin related business logic (blocks, controllers, static resources, models and resources potentially)
+    * CatalogAdmin
+    * CatalogAdminUi _recommended_
+    * CatalogBackoffice
+    
+    Storefront Ui and storefront related business logic (blocks, controllers, static resources, models and resources potentially)
+    * CatalogStorefrontUi _recommended_
+    * CatalogStorefront
+    * CatalogStore
+    
+    Cron (models and resources potentially)
+    * CatalogCron _recommended_
+    
+    CLI (models and resources potentially)
+    * CatalogCli _recommended_
+    
+    WebApi (models and resources potentially)
+    * CatalogWebApi _recommended_
+    
+    Common code (business logic used by multiple areas, models, resource modelss)
+    * CatalogCommon
```

Comment:
Please consider having this module named Catalog and have metapackage named CatalogMetapackage (or something similar).

The issue is that third-party modules will not be able to get seamless upgrade if they depend on current Catalog. We need to evaluate how realistic is that we will not bump major version of the Catalog package with this module split.

Reply:
The idea was to don't introduce a new name and 3rd party developers will have a possibility to specify multiple versions in composer (option 3 from https://github.com/magento/architecture/blob/master/design-documents/service-isolation/split-extensions.md#new-module---old-composer)